src/ctapipe/image/extractor.py

```diff
@@ -1295,6 +1296,20 @@ def __call__(
         )
 
 
+class VarianceExtractor(ImageExtractor):
+
+    """
+    Extractor that calculates
+    the variance of each waveform
+    """
+
+    def __call__(
+        self, waveforms, tel_id
+    ) -> DL1CameraContainer:
+        variance = np.nanvar(waveforms,axis=2)
+        return DL1CameraContainer(image=variance, peak_time=0, is_valid=True)
```

Comment:
I think the DL1CameraContainer should be adjusted rather than storing the variances where the charge images belong and setting peak_time to zero. This will introduce breaking changes ...

Reply:
peak_time should be `None`, not 0, which will e.g. make it so that this field is skipped when writing out.

If you need to give a value, it needs to have the correct shape

Reply:
Do we want to add a metadata field that describes the meaning of the image? Could be an enum of "charge", "variance" etc. with the default value being "charge" for backwards compatibility